### PR TITLE
Add china certificates and endpoint

### PR DESCRIPTION
**Pull request by rclee**
Brings in changes from [MGL 11860](https://github.com/mapbox/mapbox-gl-native/pull/11860) and [MGL 11845](https://github.com/mapbox/mapbox-gl-native/pull/11845)

### CODE REVIEW

File: MapboxMobileEvents/MMEAPIClient.m [hunk 1]
```diff
@@ -57,6 +57,8 @@ - (void)postEvent:(MMEEvent *)event completionHandler:(nullable void (^)(NSError
 - (void)setBaseURL:(NSURL *)baseURL {
     if (baseURL && [baseURL.scheme isEqualToString:@"https"]) {
         _baseURL = baseURL;
+    } else if ([[_applicationBundle objectForInfoDictionaryKey:@"MGLMapboxAPIBaseURL"] isEqualToString:MMEAPIClientBaseChinaAPIURL]) {
+        _baseURL = [NSURL URLWithString:MGLAPIClientChinaBaseURL];
```
Comment (1ec5) [inline MapboxMobileEvents/MMEAPIClient.m]: This should be `MMEAPIClientBaseChinaURL`.

**Comment by julianrex** [comment]
Could we make sense for `MMEAPIClientBaseChinaURL` to be `MMEAPIClientBaseChinaEventsURL` for clarity? Or does this match MGL?

**Review by lloydsheng — APPROVED** [review]
LGTM

**Comment by lloydsheng** [comment]
@rclee Just didn't see any place called `evetsManger.baseURL` or `setBaseURL` method at `release-boba` branch, how we make sure the endpoint switch logic take effect?

**Comment by rclee** [comment]
@lloydsheng since we moved the event library to be more independent much of the logic that existed in GL is now only found in the events. This is using the logic that @1ec5 created but we're just porting it over. We'll make sure to test after this merge.

**Pull request merged**